packages/world/src/modules/core/implementations/BalanceTransferSystem.sol

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import { System } from "../../../System.sol";
+import { ResourceSelector } from "../../../ResourceSelector.sol";
+import { AccessControl } from "../../../AccessControl.sol";
+import { IWorldErrors } from "../../../interfaces/IWorldErrors.sol";
+
+import { Balances } from "../tables/Balances.sol";
+
+contract BalanceTransferSystem is System, IWorldErrors {
+  using ResourceSelector for bytes32;
+
+  /**
+   * Transfer balance to another namespace in the World
+   */
+  function transferBalanceToNamespace(bytes16 fromNamespace, bytes16 toNamespace, uint256 amount) public virtual {
+    // Require caller to have access to the namespace
+    AccessControl.requireAccess(fromNamespace, _msgSender());
+
+    // Get current namespace balance
+    uint256 balance = Balances.get(fromNamespace);
+
+    // Require the balance balance to be greater or equal to the amount to transfer
+    if (amount > balance) revert InsufficientBalance(balance, amount);
+
+    // Update the balances
+    Balances.set(fromNamespace, balance - amount);
+    Balances.set(toNamespace, Balances.get(toNamespace) + amount);
+  }
+
+  /**
+   * Transfer balance out of the World
+   */
+  function transferBalanceToAddress(bytes16 fromNamespace, address toAddress, uint256 amount) public virtual {
+    // Require caller to have access to the namespace
+    AccessControl.requireAccess(fromNamespace, _msgSender());
+
+    // Get current namespace balance
+    uint256 balance = Balances.get(fromNamespace);
+
+    // Require the balance balance to be greater or equal to the amount to transfer
+    if (amount > balance) revert InsufficientBalance(balance, amount);
+
+    // Update the balances
+    Balances.set(fromNamespace, balance - amount);
+    payable(toAddress).transfer(amount);
```

Comment:
I don't know the "why" but I've seen folks prefer this approach over the above:
```solidity
(bool success, ) = toAddress.call{value: amount}("");
```

Reply:
good catch! probably because of the 2300 gas limit of `transfer` (see https://twitter.com/obatirou/status/1697167795077698038 and https://medium.com/coinmonks/solidity-transfer-vs-send-vs-call-function-64c92cfc878a)